packages/engine/src/datastore/table/pool/mod.rs

```diff
@@ -74,12 +74,26 @@ impl<P: Pool<B> + Send + Sync, B: Batch> BatchPool<B> for P {
         self.get_batches_mut().push(Arc::new(RwLock::new(batch)))
     }
 
-    fn remove(&mut self, index: usize) -> Result<BatchReadProxy<B>> {
-        BatchReadProxy::new(&self.get_batches_mut().remove(index))
+    fn remove(&mut self, index: usize) -> String {
+        let mut batch_arc = self.get_batches_mut().remove(index);
+        if let Some(rw_lock) = Arc::get_mut(&mut batch_arc) {
+            // This can't deadlock as we just checked that the Arc owning this RwLock is unique
+            let batch = rw_lock.write();
+            batch.get_batch_id().to_string()
```

Comment:
```suggestion
            rw_lock.read().get_batch_id().to_string()
```

Reply:
Have turned it into one line, but keeping it as `write` because it's a little clearer that it's an exclusive lock on the RwLock not just the arc